pytorch_lightning/trainer/trainer.py

```diff
@@ -2471,6 +2474,28 @@ def _active_loop(self) -> Optional[Union[FitLoop, EvaluationLoop, PredictionLoop
     Logging properties
     """
 
+    @property
+    def logger(self) -> Optional[LightningLoggerBase]:
+        return self._logger
+
+    @logger.setter
+    def logger(self, new_logger: Optional[LightningLoggerBase]) -> None:
+        self._logger = new_logger
```

Comment:
The `loggers` setter will set this, no?

```suggestion
```

Reply:
That's right. I've had to rewrite the setter to handle a special case but now the `loggers` setter does all the work.